apps/meteor/client/views/room/hooks/useUserInfoActions/actions/useRemoveUserAction.tsx

```diff
@@ -16,9 +17,13 @@ import { getRoomDirectives } from '../../../lib/getRoomDirectives';
 export const useRemoveUserAction = (user: Pick<IUser, '_id' | 'username'>, rid: IRoom['_id'], reload?: () => void): Action | undefined => {
 	const t = useTranslation();
 	const room = useUserRoom(rid);
+	const currentUser = useUser();
 	const { _id: uid } = user;
 
-	const userCanRemove = usePermission('remove-user', rid);
+	const hasPermissionToRemove = usePermission('remove-user', rid);
+	const userCanRemove = isRoomFederated(room as Partial<IRoom>)
+		? Federation.canEdit(currentUser as IUser, room as IRoom)
```

Comment:
did you have to add the type castings because they can be `null`/`undefined`? what if they're really `undefined`, then your code could break =\ you should properly validate the values instead of type casting
```suggestion
	const userCanRemove = isRoomFederated(room)
		? Federation.canEdit(currentUser, room)
```